torch/functional.py

```diff
@@ -331,9 +331,33 @@ def meshgrid(*tensors: Union[Tensor, List[Tensor]]) -> Tuple[Tensor, ...]:
         return _meshgrid(*tensors)
 else:
     def meshgrid(*tensors):
-        r"""Take :math:`N` tensors, each of which can be either scalar or 1-dimensional
-        vector, and create :math:`N` N-dimensional grids, where the :math:`i` :sup:`th` grid is defined by
-        expanding the :math:`i` :sup:`th` input over dimensions defined by other inputs.
+        r"""Creates grids of coordinates specified by the input tensors.
+
+        Take :math:`N` tensors :math:`T_0, T_1, \ldots, T_N`, each of
+        which can be a scalar* or a vector. Produces :math:`N`
+        N-dimensional tensors, each of shape :math:`(|T_0|, |T_1|
+        \ldots |T_n|)`. Each tensor output :math:`G_i` is :math:`T_i`
+        in the :math:`i` :sup:`th` dimension, expanded over the other
+        dimensions.
+
+        This operation is conceptually similar to the cartesian
+        product, except the output is collected across multiple
+        tensors, rather than a tensor of vectors.
+
+        * Note that input scalars are treated equivalently to vectors
+          of a single element.
+
+        .. warning::
+
+            This behaves differently from `numpy.meshgrid`.
```

Comment:
Cool. Although depending on when this PR goes in this may need to change because meshgrid will support 'xy' indexing, too